src/main/java/io/supertokens/webserver/api/core/HelloAPI.java

```diff
@@ -55,31 +57,30 @@ protected boolean checkAPIKey(HttpServletRequest req) {
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        handleRequest(resp);
+        handleRequest(req, resp);
     }
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        handleRequest(resp);
+        handleRequest(req, resp);
     }
 
     @Override
     protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        handleRequest(resp);
+        handleRequest(req, resp);
     }
 
     @Override
     protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        handleRequest(resp);
+        handleRequest(req, resp);
     }
 
-    private void handleRequest(HttpServletResponse resp) throws IOException, ServletException {
-        // TODO: need to get the right storage based on TenantIdentifier input.
-        Storage storage = StorageLayer.getBaseStorage(main);
+    private void handleRequest(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         try {
-            storage.getKeyValue(new TenantIdentifier(null, null, null), "Test");
+            TenantIdentifierWithStorage tenantIdentifierWithStorage =  getTenantIdentifierWithStorageFromRequest(req);
+            tenantIdentifierWithStorage.getStorage().getKeyValue(tenantIdentifierWithStorage, "Test");
```

Comment:
Actually, since this API is open to all (does not require api key), maybe we should make it so that it does the above only when an API key is provided (api key fetched based on the scope of the request). If api key is missing, it will only query the db of the current tenant (which is what you have implemented).

Or maybe we can add another endpoint that always requires the api key and does the checks for all storages based on the scope of the request? To discuss